frontend/src/metabase/dashboard/containers/DashboardApp/DashboardApp.unit.spec.jsx

```diff
@@ -0,0 +1,181 @@
+import React from "react";
+import { Route } from "react-router";
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+import {
+  within,
+  screen,
+  renderWithProviders,
+  waitForElementToBeRemoved,
+  fireEvent,
+} from "__support__/ui";
+import DashboardApp from "metabase/dashboard/containers/DashboardApp";
+import { BEFORE_UNLOAD_UNSAVED_MESSAGE } from "metabase/hooks/use-before-unload";
+import {
+  createMockCard,
+  createMockCollection,
+  createMockCollectionItem,
+  createMockDashboard,
+  createMockDatabase,
+  createMockTable,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import { createMockDashboardState } from "metabase-types/store/mocks";
+import {
+  setupCardsEndpoints,
+  setupCollectionItemsEndpoint,
+  setupCollectionsEndpoints,
+  setupDashboardEndpoints,
+  setupDatabasesEndpoints,
+  setupSearchEndpoints,
+  setupTableEndpoints,
+} from "__support__/server-mocks";
+import { createMockEntitiesState } from "__support__/store";
+import { callMockEvent } from "__support__/events";
+
+const TEST_DASHBOARD = createMockDashboard();
+const TEST_COLLECTION = createMockCollection();
+
+const TEST_DATABASE_WITH_ACTIONS = createMockDatabase({
+  settings: { "database-enable-actions": true },
+});
+
+const TEST_COLLECTION_ITEM = createMockCollectionItem({
+  collection: TEST_COLLECTION,
+  model: "dataset",
+});
+
+const TEST_CARD = createMockCard();
+
+const TEST_TABLE = createMockTable();
+
+async function setup({ user = createMockUser() }) {
+  setupDatabasesEndpoints([TEST_DATABASE_WITH_ACTIONS]);
+  setupDashboardEndpoints(createMockDashboard(TEST_DASHBOARD));
+  setupCollectionsEndpoints([]);
+  setupCollectionItemsEndpoint(TEST_COLLECTION);
+  setupSearchEndpoints([TEST_COLLECTION_ITEM]);
+  setupCardsEndpoints([TEST_CARD]);
+  setupTableEndpoints([TEST_TABLE]);
+
+  fetchMock.get("path:/api/bookmark", []);
```

Comment:
I think there are existing helpers for that in `server-mocks`, if not, please add missing ones there

Reply:
Yep, found them 😄